member-service/src/main/java/com/hedvig/memberservice/aggregates/MemberAggregate.kt

```diff
@@ -0,0 +1,588 @@
+
+    @Autowired
+    lateinit var objectMapper: ObjectMapper
+
+    @Autowired
+    lateinit var uuidGenerator: UUIDGenerator
+
+    lateinit var status: MemberStatus
+    lateinit var member: Member
+    var trackingId: UUID? = null
+    lateinit var latestBisnodeInformation: BisnodeInformation
+
+    @CommandHandler
+    constructor(command: CreateMemberCommand) : this() {
+        apply(MemberCreatedEvent(command.memberId, MemberStatus.INITIATED))
+            .andThenApply {
+                if (command.acceptLanguage != null && !command.acceptLanguage.isEmpty()) {
+                    return@andThenApply AcceptLanguageUpdatedEvent(command.memberId, command.acceptLanguage)
+                }
+                null
+            }
+    }
+
+    @CommandHandler
+    fun authAttempt(command: SwedishBankIdAuthenticationAttemptCommand) {
+        var applyChain: ApplyMore? = null
+        val bankIdAuthResponse = command.bankIdAuthResponse
+        if (status == MemberStatus.INITIATED || status == MemberStatus.ONBOARDING) {
+            // Trigger fetching of bisnode data.
+            val ssn = bankIdAuthResponse.sSN
+            applyChain = apply(SSNUpdatedEvent(id, ssn, SSNUpdatedEvent.Nationality.SWEDEN))
+            // -- Tracking id generation for the new member id
+            generateTrackingId()
+            applyChain = getBirthdateFromSwedishSsnAndApplyEvent(ssn, applyChain)
+            applyChain = try {
+                getPersonInformationFromBisnode(applyChain, ssn)
+            } catch (ex: RuntimeException) {
+                logger.error("Caught exception calling bisnode for personalInformation", ex)
+                applyChain!!.andThenApply {
+                    NameUpdatedEvent(
+                        id,
+                        formatName(bankIdAuthResponse.givenName),
+                        formatName(bankIdAuthResponse.surname))
+                }
+            }
+            if (status == MemberStatus.INITIATED) {
+                applyChain = applyChain!!.andThenApply { MemberStartedOnBoardingEvent(id, MemberStatus.ONBOARDING) }
+            }
+        }
+        val authenticatedEvent = MemberAuthenticatedEvent(id, bankIdAuthResponse.referenceToken)
+        if (applyChain != null) {
+            applyChain.andThenApply(Supplier<Any> { authenticatedEvent })
+        } else {
+            apply(authenticatedEvent)
+        }
+    }
+
+    private fun getBirthdateFromSwedishSsnAndApplyEvent(ssn: String?, @Nullable applyChain: ApplyMore?): ApplyMore? {
+        var applyChain = applyChain
+        val birthDate = getBirthdateFromSwedishSsn(ssn)
+        if (birthDate != null) {
+            if (applyChain != null) {
+                applyChain = applyChain.andThenApply(
+                    Supplier<Any> { BirthDateUpdatedEvent(id, birthDate) }
+                )
+                return applyChain
+            }
+            apply(BirthDateUpdatedEvent(id, birthDate))
```

Comment:
Don't we want this one back maybe?
```suggestion
            return apply(BirthDateUpdatedEvent(id, birthDate))
```

Reply:
no if applyChain is not null it will be returned other wise we do not care :slightly_smiling_face: